Automation Examples/020.md

```diff
@@ -0,0 +1,161 @@
+
+Right click the image element and see that the "src" attribute stores a link to it. The below JavaScript code returns a link to the first image. You can run it in the console to check it. 
+
+```js
+document.getElementsByClassName("lazyautosizes lazyloaded")[0].getAttribute("src")
+```
+
+So let's wait for this element using the below G1ANT code.
+
+```G1ANT
+selenium.waitforvalue ‴return document.getElementsByClassName("lazyautosizes lazyloaded").length > 0‴ expectedvalue true
+```
+
+After right clicking the product name and price, you may see that geting those values is as easy as below. 
+
+```G1ANT
+selenium.runscript ‴return document.getElementsByClassName("productSketch-name-link")[0].innerText‴ result ♥productName
+list.add ♥products toadd ♥productName
+selenium.runscript ‴return document.getElementsByClassName("regularPrice")[0].innerText‴ result ♥price
+list.add ♥prices toadd ♥price
+```
+
+#### 4. Donwload the first image
+
+In order to do this, add "https:" prefix to the "src" and use file.download command.
+
+```G1ANT
+♥imageSrc = ⊂"https:" + ♥imageSrc⊃
+file.download ♥imageSrc filename ♥basePath\1.png
+```
+
+#### 5. Put all above steps in a loop
+
+This is what you should get:
+
+```G1ANT
+for ♥i from 0 to 19
+    selenium.runscript ‴return document.getElementsByClassName("lazyautosizes lazyloaded")[♥i].getAttribute("src")‴ result ♥imageSrc
+    ♥imageSrc = ⊂"https:" + ♥imageSrc⊃
+    file.download ♥imageSrc filename ♥basePath\⊂♥i+1⊃.png
+    selenium.runscript ‴return document.getElementsByClassName("productSketch-name-link")[♥i].innerText‴ result ♥productName
+    list.add ♥products toadd ♥productName
+    selenium.runscript ‴return document.getElementsByClassName("regularPrice")[♥i].innerText‴ result ♥price
+    list.add ♥prices toadd ♥price
+end for
+```
+
+#### 6. After running this script, you may notice that it fails after some time
+
+It does so because webpage isn't fully loaded and some of the elements are not yet present so the script can't continue. To load more elements, you should scroll. Insert below line at the end of the loop to prevent this.
+
+```G1ANT
+    selenium.runscript ‴window.scrollBy(0, 200)‴
+```
+
+#### 7. Close the browser and open excel template
+
+```G1ANT
+selenium.close
+
+excel.open ‴♥basePath\Images Template.xlsx‴
+```
+
+#### 8. Create a loop, insert images, products and prices
+
+After pressing alt in the excel application, some of the letters appear which indicates what shortcut to use if you want to open some specific option from the menu. In this case, we need to press "n" and "p" to insert an image. Using keyboard to insert long texts letter by letter is sometimes not the fastest option so let's use clipboard for that and Ctrl+V shortuct to paste it in the "Save As" window. Then, set values for product name and price. Lastly, go the first column, next row by using Home and Down keys.
+
+```G1ANT
+for ♥i from 1 to 20
+    keyboard ⋘alt⋙np
```

Comment:
⋘alt⋙np
shouldn't it be more like ⋘alt+n+p⋙?

Reply:
No. To access something from the menu, you need to press alt so that letters are underlined and focused, then "n" to open "Insert" tab, then "P". It is crucial because sometimes to access some option, you have two letters next to each other (which means that they have to be pressed at the same time). See next example where in order to open "Format" tab, you need to press "J+P"